### ISSUE TITLE
Failed to deserialize simple long milliseconds, nano seconds are disabled.

### ISSUE THREAD

**Issue by vanyne** [FasterXML/jackson-datatype-jsr310#51]
Mapper configuration:

``` java
 ObjectMapper mapper = new ObjectMapper();
            mapper.setSerializationInclusion(Include.NON_NULL);
            mapper.findAndRegisterModules();
            mapper.disable(DeserializationFeature.READ_DATE_TIMESTAMPS_AS_NANOSECONDS);
            mapper.disable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS);

            mapper.enable(DeserializationFeature.READ_ENUMS_USING_TO_STRING);
            mapper.enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);
```

Error Stack Trace:

```
Caused by: java.time.format.DateTimeParseException: Text '1449163142741' could not be parsed at index 0
    at java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:1947)
    at java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1775)
    at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:151)
    at com.fasterxml.jackson.datatype.jsr310.deser.InstantDeserializer.deserialize(InstantDeserializer.java:45)
    at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:520)
    at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:95)
    at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:256)
```

**Comment by cowtowncoder** [comment]
Which version? With what code, input?

**Comment by vanmyne** [comment]
We were using 
<dependency>
		    <groupId>com.fasterxml.jackson.datatype</groupId>
		    <artifactId>jackson-datatype-jsr310</artifactId>
		    <version>2.6.1</version>
</dependency>

Code which is trying to read from mapper:
            ObjectMapper mapper = new ObjectMapper();
            mapper.setSerializationInclusion(Include.NON_NULL);
            mapper.findAndRegisterModules();
            mapper.disable(DeserializationFeature.READ_DATE_TIMESTAMPS_AS_NANOSECONDS);
            mapper.disable(SerializationFeature.WRITE_DATE_TIMESTAMPS_AS_NANOSECONDS);

            mapper.enable(DeserializationFeature.READ_ENUMS_USING_TO_STRING);
            mapper.enable(SerializationFeature.WRITE_ENUMS_USING_TO_STRING);

            mapper.readValue(jsonString, <ClassType>);

Input <jsonString> : 
 "processorConfirmation": [
                    {
                        "processorName": "PROCESSOR1",
                        "proctxn": "aaea43b4402a4a589d36b35155d0447d",
                        "procType": "TEST1"
                    },
                    {
                        "confirmationCode": "RERQERR",
                        "confirmationTime": "1449163142741",
                        "processorName": "PROCESSOR2",
                        "processorTransactionType": "TEST2"
                    }
                ]
            }

It is failing while trying to parse the element confirmationTime.

**Comment by cowtowncoder** [comment]
@vanyne  Thank you. One more missing piece: definition of actual `ClassType` in question; I don't know which JSR-310 date type is involved here.

**Comment by vanyne** [comment]
@cowtowncoder Thanks for responding. Sorry for the missing piece.
Here is the JAXB generated class with the adapter

Class Type :
@XmlAccessorType(XmlAccessType.FIELD)
@XmlType(name="processorConfirmationType", propOrder={"confirmationCode", "processorName", "confirmationTime", "processorTransactionType", "processorTransactionID", "futureRetrievalKeys", "processorParameters"})
public class ProcessorConfirmationType implements Serializable {
	
	 private static final long serialVersionUID = 12343L;

	   @XmlElement(required=true)
       protected String confirmationCode;

       @XmlElement(required=true)
       protected String processorName;

       @XmlJavaTypeAdapter(Adapter1.class)
       @XmlSchemaType(name="dateTime")
       protected Instant confirmationTime;     //Failing piece of code

       @XmlElement(required=true)
       protected String processorTransactionType;

       ........
}

**Comment by cowtowncoder** [comment]
Not sure why `Instant` values are serialized as Strings, but I think #16 fix should apply here too.
Perhaps this was XML output, not JSON?

**Comment by cowtowncoder** [comment]
Assuming this is same as #16, closing as dup. Fix will be in 2.6.6 and 2.7.1 (latter to be out sooner).

**Comment by vanyne** [comment]
2.7.1. fixed the issue. I haven't seen any deserialization error with that version of JSR310.
Thank you!
